File: scratch/updateRouter.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(
  'c:', 'Users', 'localhost', 'Documents', 'Enrollpro',
  'server', 'src', 'features', 'students', 'students.router.ts'
);

let content = fs.readFileSync(filePath, 'utf8');

content = content.replace(
  /verifyPsa,/,
  "verifyPsa,\n  updateLrn,\n  markDropout,\n  markTransferredOut,"
);

const routes = `
// Lifecycle & LRN Management
router.post(
  "/:id/lifecycle/dropout",
  authorize("HEAD_REGISTRAR", "SYSTEM_ADMIN"),
  markDropout
);

router.post(
  "/:id/lifecycle/transfer-out",
  authorize("HEAD_REGISTRAR", "SYSTEM_ADMIN"),
  markTransferredOut
);

router.post(
  "/:id/lrn",
  authorize("HEAD_REGISTRAR", "SYSTEM_ADMIN"),
  updateLrn
);

export default router;
`;

content = content.replace(/export default router;/, routes);

fs.writeFileSync(filePath, content, 'utf8');
console.log("Updated students.router.ts");

File: scratch/update_router.py
import re

ROUTER_PATH = r'c:\Users\localhost\Documents\Enrollpro\client\src\router\index.tsx'

with open(ROUTER_PATH, 'r', encoding='utf-8') as f:
    content = f.read()

content = content.replace('path: "/bosy",', 'path: "/continuing-learners",')

with open(ROUTER_PATH, 'w', encoding='utf-8') as f:
    f.write(content)

print("Router updated.")
